workflow/controller/controller.go

```diff
@@ -765,3 +766,8 @@ func (wfc *WorkflowController) getMetricsServerConfig() (metrics.ServerConfig, m
 
 	return metricsConfig, telemetryConfig
 }
+
+func (wfc *WorkflowController) isArchivable(wf *wfv1.Workflow) bool {
+	return wfc.archiveLabelSelector.Matches(labels.Set(wf.Labels))
```

Comment:
There is a `nil` exception bug waiting to happen here if persistence is not enabled and `archiveLabelSelector` is not set...

This is one of the issue with using multiple sources of truth.

Reply:
good catch. I will fix it. 
`wfc.archiveLabelSelector` didn't initialize with `labels.Everything()`